stubs/docutils/docutils/parsers/rst/__init__.pyi

```diff
@@ -1,34 +1,67 @@
 from _typeshed import Incomplete
+from collections.abc import Callable, Sequence
 from typing import Any, ClassVar, Literal
+from typing_extensions import TypeAlias
 
-from docutils import parsers
-from docutils.parsers.rst import states
+from docutils import nodes, parsers
+from docutils.parsers.rst.states import Inliner, RSTState, RSTStateMachine
+from docutils.statemachine import StringList
+from docutils.transforms import Transform
 
 class Parser(parsers.Parser):
+    settings_spec: ClassVar[Incomplete]
     config_section_dependencies: ClassVar[tuple[str, ...]]
     initial_state: Literal["Body", "RFC2822Body"]
-    state_classes: Any
-    inliner: Any
-    def __init__(self, rfc2822: bool = False, inliner: Incomplete | None = None) -> None: ...
+    state_classes: Sequence[type[RSTState]]
+    inliner: Inliner
+    def __init__(self, rfc2822: bool = False, inliner: Inliner | None = None) -> None: ...
+    def get_transforms(self) -> list[type[Transform]]: ...
+    def parse(self, inputstring: str, document: nodes.document) -> None: ...
 
 class DirectiveError(Exception):
-    level: Any
+    level: int
     msg: str
-    def __init__(self, level: Any, message: str) -> None: ...
+    def __init__(self, level: int, message: str) -> None: ...
 
 class Directive:
+    required_arguments: int
+    optional_arguments: int
+    final_argument_whitespace: bool
+    option_spec: dict[str, Callable[[str], Any]] | None
+    has_content: bool
+    name: str
+    arguments: list[str]
+    options: dict[str, Any]
+    content: StringList
+    lineno: int
+    content_offset: int
+    block_text: str
+    state: RSTState
+    state_machine: RSTStateMachine = ...
     def __init__(
         self,
         name: str,
-        arguments: list[Any],
+        arguments: list[str],
         options: dict[str, Any],
-        content: list[str],
+        content: StringList,
         lineno: int,
         content_offset: int,
         block_text: str,
-        state: states.RSTState,
-        state_machine: states.RSTStateMachine,
+        state: RSTState,
+        state_machine: RSTStateMachine,
     ) -> None: ...
-    def __getattr__(self, name: str) -> Incomplete: ...
+    def run(self) -> Sequence[nodes.Node]: ...
```

Comment:
Here, we could keep the `list` because we usually return a new list, so the function should be pure. With a list, we can write something like

```
ret = super().run()
ret.extend(...)
return ret
```

in a subclass and this pattern might be quite common. Or we can use `insert` for instance. As such, keeping a mutable sequence of the correct type should be fine, although we lose the covariance of sequences (but I don't think the covariance is needed and mutability is more interesting).

Reply:
i'm unsure of the best approach here.
I think the covariance property is more important since if it's typed as `list[Node]` and you return `[nodes.Subclass, nodes.OtherSubclass]` that's going to be an error, right?

Also it's been typed as `Sequence` in docutils-stubs for years

and finally, if you really need a list you can always write:

```python
ret = list(super().run())
ret.extend(...)
return ret
```

I'm open to being convinced though

Reply:
Mmh. Ok, the covariance is actually more important now that you think about it. 

Let's keep it like that.